DaemonRenderer/Source/Src/Windowing/Window.cpp

```diff
@@ -24,20 +24,52 @@
 
 #include "Windowing/Window.hpp"
 
+#ifdef DAEMON_OS_WINDOWS
+    #define GLFW_EXPOSE_NATIVE_WIN32
+#endif
+
+#include <GLFW/glfw3native.h>
+
+#undef CreateWindow
+
+#include "Debug/Logging/Logger.hpp"
+
 USING_DAEMON_NAMESPACE
 
-#pragma region Constructors and Destructor
+#pragma region Constructors
 
-Window::Window(WindowParams const& in_params) noexcept:
-    m_handle    {nullptr},
+Window::Window(Logger* in_logger, WindowParams const& in_params) noexcept:
+    m_logger    {in_logger},
     m_name      {in_params.name}
 {
-    SetupWindow   (in_params);
+    CreateWindow  (in_params);
     SetupCallbacks();
 }
 
+Window::Window(Window&& in_move) noexcept:
+    m_logger                {in_move.m_logger},
```

Comment:
Both the logger and the handle can be moved too here since they are pointers, the actual objects will remain untouched

Reply:
A pointer is trivially copyable so moving it is less optimized.